### ISSUE TITLE
Global reference style links

### ISSUE THREAD

**Issue by dsm-72** [jupyterlab/jupyterlab#14260]
### Problem

I  like using reference style markdown links e.g.

```
some text about a [thing][link to something]

[link to something]: https://www.link-to-something.com
```

however it seems that Jupyter notebooks don't support this, let alone letting me define all of my links in one markdown cell and referencing them in other cells.

### Proposed Solution

Either a magic command `%global_links` or a different parsing method `[$global_link]: https:...` to indicate urls that should be used everywhere

**Comment by krassowski** [comment]
> however it seems that Jupyter notebooks don't support this,

unless I misunderstood, the above statement appears incorrect as seen on the recording below:

![it-works](https://user-images.githubusercontent.com/5832902/226910872-c74e68b7-85f7-49ce-8a45-ec12d66fa415.gif)

@dsm-72 what version of JupyterLab are you using?

> let alone letting me define all of my links in one markdown cell and referencing them in other cells.

yes, this is not currently supported and a valid enhancement request.

**Comment by dsm-72** [comment]
@krassowski for the note supporting reference links I meant notebook wide per the second one. Although for those who use `nbdev` reference style links do not get rendered correctly, but that is a separate issue:

https://user-images.githubusercontent.com/100158155/226913654-cc17642f-a95e-4d6e-add0-3d68684bfaca.mov

**Comment by krassowski** [comment]
@dsm-72 this is not Jupyter Notebook nor JupyterLab but a third-party application which appears similar to VSCode from Microsoft. These are not developed, maintained nor supported by Jupyter team.

**Comment by dsm-72** [comment]
@krassowski 

re vscode: same behavior in Jupyter Notebooks. Just was faster to make an example this way

re nbdev: yes. I was only pointing that out

**Comment by dsm-72** [comment]
@andrii-i @JasonWeill  any updates? How is the JEP [PR](https://github.com/jupyter/enhancement-proposals/pull/99).  @tonyfast  any ideas? Any way I can help move this along?

**Comment by krassowski** [comment]
JupyterLab currently uses `marked.js` to render markdown so you would probably want to look how to feed that Markdown parser with information extracted from other cells. However since this would likely be a hard problem (in unconstrained naive version each cell could contribute link info for each cell, leading to n^2 complexity if implemented naively) this may require quite some work on the side of caching the parser data for other cells to avoid re-parsing.

Would willing to contribute a PR along this lines?

**Comment by dsm-72** [comment]
@krassowski possibly, although I am fairly unfamiliar with the code base. Would it be simpler to use or create a "magic" expression to solve it then? 


```python
%global_references
 [link 1]: www.something.com
 [link 2]: another.com
```

then you only have to check one cell?

**Comment by krassowski** [comment]
I don't think so. Any kind of magics would require working with kernels which further increases the complexity and reduces likelihood this would be merged into core codebase. But if you can create an extension that solves your need this way - why not.

**Comment by dsm-72** [comment]
@krassowski is there like a dedicated plugin system like conda plugins? or can you point me to something simple where I can  wrangle something for this?
